libgnucash/engine/gnc-optiondb.cpp

```diff
@@ -547,7 +547,7 @@ GncOptionDB::load_from_kvp(QofBook* book) noexcept
                             set_double();
                             break;
                         case KvpValue::Type::INT64:
-                            option.set_value(kvp->get<int64_t>());
+                            option.set_value(static_cast<int>(kvp->get<int64_t>()));
```

Comment:
I believe this is what cause the counters not to be populated after the first save, when the doubles have been converted to int. What is less clear is since this is generic kvp code, what else is stored as int64_t? If other things are stored as int64_t and need to be loaded as such (not as int) then I suppose we need to do here the same thing that you did for the double above, meaning testing if we are dealing with counters and if so retrieve as int, otherwise retrieve as int64_t.

Reply:
@dawansv Indeed, good catch.

This code fortunately applies only to file properties. The only reason this would break is if there's a GncOptionValue<int64_t> in there and I'm pretty sure there isn't.

Reply:
Oh good yes now I see it's only called from `gnc_book_options_dialog_cb` via `gnc_option_db_load`, and also from `assistant_insert_book_options_page` but that's the assistant populating the file properties so going to the same place. 
I just tested some of the other properties in that dialog; there aren't many that are numbers, and they all seem to work fine, so as you say I don't think we have any other int64_t options in there.